x-pack/filebeat/input/httpjson/request.go

```diff
@@ -163,21 +181,160 @@ func newRequester(
+				if err != nil {
+					return fmt.Errorf("failed to execute http client.Do: %w", err)
+				}
+				// store data according to response type
+				if i+1 == len(r.requestFactory) && len(ids) != 0 {
+					finalResps = append(finalResps, httpResp)
+				} else {
+					intermediateResps = append(intermediateResps, httpResp)
+				}
+			}
+			rf.url = urlCopy
+
+			if i+1 == len(r.requestFactory) {
+				split = r.responseProcessor.split
+				r.responseProcessor.split = nil
+				n, err = r.processAndPublishEvents(0, stdCtx, trCtx, publisher, finalResps)
+				if err != nil {
+					return err
+				}
+				r.responseProcessor.split = split
+				continue
+			} else {
+				ids, err = r.getIdsFromResponses(intermediateResps, r.requestFactory[i+1].replace)
+				if err != nil {
+					return err
+				}
+				split = r.responseProcessor.split
+				r.responseProcessor.split = nil
+				n, err = r.processAndPublishEvents(1, stdCtx, trCtx, publisher, intermediateResps)
+				if err != nil {
+					return err
+				}
+				r.responseProcessor.split = split
+			}
+		} else {
+			// perform and store regular call responses
+			httpResp, err = rf.collectResponse(stdCtx, trCtx, r, publisher)
+			if err != nil {
+				return fmt.Errorf("failed to execute http client.Do: %w", err)
+			}
+			if len(r.requestFactory) <= 1 {
+				finalResps = append(finalResps, httpResp)
+				n, err = r.processAndPublishEvents(0, stdCtx, trCtx, publisher, finalResps)
+				if err != nil {
+					return err
+				}
+				continue
+			} else {
+				intermediateResps = append(intermediateResps, httpResp)
+				ids, err = r.getIdsFromResponses(intermediateResps, r.requestFactory[i+1].replace)
+				if err != nil {
+					return err
+				}
+				n, err = r.processAndPublishEvents(1, stdCtx, trCtx, publisher, intermediateResps)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	httpResp.Body.Close()
+	r.log.Infof("request finished: %d events published", n)
+
+	return nil
+}
+
+// getIdsFromResponses returns ids from responses
+func (r *requester) getIdsFromResponses(intermediateResps []*http.Response, replace string) ([]string, error) {
+	var b []byte
+	var ids []string
+	var err error
+	// collect ids from all responses
+	for _, resp := range intermediateResps {
+		if resp.Body != nil {
+			b, err = ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("error while reading response body: %v", err)
+			}
+		}
+		// Restore the io.ReadCloser to its original state
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(b))
+		// get replace values from collected json
+		ids, err = parse(string(b), replace)
+		if err != nil {
+			return nil, fmt.Errorf("error while getting keys: %v", err)
+		}
+	}
+	return ids, nil
+}
+
+// processAndPublishEvents process and publish events based on response type
+func (r *requester) processAndPublishEvents(publish int, stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher, finalResps []*http.Response) (int, error) {
```

Comment:
By convention `context.Context` should be the first parameter, I suggest to refactor this method signature.